src/token_contract/src/main.nr

```diff
@@ -1,45 +1,30 @@
 mod types;
 mod test;
 
+use crate::types::token_utils::token;
 use dep::aztec::macros::aztec;
 
-// Minimal token implementation that supports `AuthWit` accounts.
-// The auth message follows a similar pattern to the cross-chain message and includes a designated caller.
-// The designated caller is ALWAYS used here, and not based on a flag as cross-chain.
-// message hash = H([caller, contract, selector, ...args])
-// To be read as `caller` calls function at `contract` defined by `selector` with `args`
-// Including a nonce in the message hash ensures that the message can only be used once.
+#[token]
 #[aztec]
 pub contract Token {
 
     use dep::compressed_string::FieldCompressedString;
     use dep::std::ops::{Add, Sub};
 
     use dep::aztec::{
-        context::{PrivateCallInterface, PrivateContext},
         encrypted_logs::log_assembly_strategies::default_aes128::note::encode_and_encrypt_note,
```

Comment:
the `[token]` tag needs an `encode_and_encrypt_note` available at the main contract to compile, this is the way how the token developer decides on which strategy to use

to achieve this:
- encryption methods are on Token code (not TokenLib)
- `encode_and_encrypt_note` is optimistically assumed to exist (shitty DX tho)
- main contract declares some `encode_and_encrypt_note`
- at comptime it gets injected all together